gigascience/gigadb-website
# Accessibility tweaks on datasetSubmission/upload page

# Pull request for issue: #1743

This is a pull request for the following functionalities:

* Updated form in http://gigadb.gigasciencejournal.com/datasetSubmission/upload to improve accessibility

## How to test?

Navigate form with keyboard and verify that all interactive controls receive focus and form cannot be submitted without first filling fields

## How have functionalities been implemented?

* removed disabled attributes so keyboard focus is possible even with incomplete fields
* keep checkbox error hidden by default
* programmatically set aria-describedby for checkbox only when error is visible

Note: I removed the logic that forced the user to check the checkbox before updating a file. If this is explicitly needed it can be updated. Since the form cannot be submitted unless both file and checkbox are fulfilled anyway, it doesn't seem necessary

## Any issues with implementation?

--

## Any changes to automated tests?

--
